packages/pages/src/upgrade/migrate_config.ts

```diff
@@ -2,58 +2,116 @@ import path from "path";
 import fs from "fs";
 import yaml from "yaml";
 
-const migrateServing = async (
+function readJsonSync(jsonPath: string): any {
+  if (!fs.existsSync(jsonPath)) {
+    return null;
+  }
+  try {
+    return JSON.parse(fs.readFileSync(jsonPath, "utf-8"));
+  } catch (e) {
+    console.error(`unable to parse ${jsonPath}: ${(e as Error).message}`);
+    return null;
+  }
+}
+
+function writeYamlSync(configYamlPath: string, target: string, data: any) {
+  const yamlDoc = yaml.parseDocument(fs.readFileSync(configYamlPath, "utf-8"));
+  yamlDoc.set(target, data);
+  fs.writeFileSync(configYamlPath, yaml.stringify(yamlDoc));
+}
+
+const migrateCiJson = async (
   configYamlPath: string,
   sitesConfigPath: string
 ) => {
-  const servingPath = path.resolve(sitesConfigPath, "serving.json");
-  if (fs.existsSync(servingPath)) {
-    const servingJson = JSON.parse(fs.readFileSync(servingPath, "utf-8"));
-    const displayUrlPrefix = servingJson.displayUrlPrefix;
-    if (displayUrlPrefix !== null) {
-      console.info(
-        "migrating reverse proxy info from serving.json to config.yaml"
-      );
-      const yamlDoc = yaml.parseDocument(
-        fs.readFileSync(configYamlPath, "utf-8")
-      );
-      yamlDoc.set("reverseProxy", {
-        displayUrlPrefix: displayUrlPrefix,
-      });
-      fs.writeFileSync(configYamlPath, yaml.stringify(yamlDoc));
-    }
+  const ciJson = readJsonSync(path.resolve(sitesConfigPath, "ci.json"));
+  if (ciJson !== null) {
+    ciJson.artifactStructure = null;
+    console.info("migrating site config from ci.json to config.yaml");
+    writeYamlSync(configYamlPath, "ciConfig", ciJson);
   }
 };
 
-const migrateGlobalData = async (
+const migrateLocales = async (
   configYamlPath: string,
   sitesConfigPath: string
 ) => {
-  const siteStreamPath = path.resolve(sitesConfigPath, "site-stream.json");
-  if (fs.existsSync(siteStreamPath)) {
-    const sitesJson = JSON.parse(fs.readFileSync(siteStreamPath, "utf-8"));
+  const featuresJson = readJsonSync(
+    path.resolve(sitesConfigPath, "features.json")
+  );
+  if (!!featuresJson && !!featuresJson.locales) {
+    console.info("migrating locales from features.json to config.yaml");
+    writeYamlSync(configYamlPath, "locales", featuresJson.locales);
+  }
+};
+
+const migrateSiteStream = async (
+  configYamlPath: string,
+  sitesConfigPath: string
+) => {
+  const sitesJson = readJsonSync(
+    path.resolve(sitesConfigPath, "site-stream.json")
+  );
+  if (sitesJson !== null) {
     console.info("migrating global data from site-stream.json to config.yaml");
-    const yamlDoc = yaml.parseDocument(
-      fs.readFileSync(configYamlPath, "utf-8")
+    writeYamlSync(configYamlPath, "globalData", sitesJson);
+  }
+};
+
+const migrateRedirects = async (
+  configYamlPath: string,
+  sitesConfigPath: string
+) => {
+  const redirectsPath = path.resolve(sitesConfigPath, "redirects.csv");
```

Comment:
We'll need to check with the PMs and Joe about keeping redirects.csv. This file can be large so it doesn't really make sense to jam it all into the yaml. The second part of the question is if this file still needs to live in sites-config or does it now live at the root?

Or even more, is the file path referenced in the config.yaml?

Reply:
Current implementation is to move redirects.csv next to config.yaml